service/dap/server.go

```diff
@@ -840,9 +865,10 @@ func (s *Server) onAttachRequest(request *dap.AttachRequest) {
 		s.config.Debugger.AttachPid = int(pid)
 		s.setLaunchAttachArgs(request)
 		var err error
+		s.mu.Lock()
+		defer s.mu.Unlock()
```

Comment:
This lock lives beyond the if block (so affect line 884-885) which isn't obvious. 
Consider either narrowing the scope of the lock, so lock is released before going out of the if block, or
handle if mode != "local" case first 

if mode != "local" {
  s.sendErrorResponse(...)
  return
}
// mode == "local"
...

Reply:
Will do. I am actually not happy how much logic ends up inside of lock in general :(